Seed.Domain/Services/Interfaces/IBaseService.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Seed.Data.Models;
+
+namespace Seed.Domain.Services.Interfaces
+{
+    public interface IBaseService<T> where T : BaseEntity
+    {
+        /// <summary>
+        /// Gets a user
+        /// </summary>
+        /// <param name="guid">Id of the entity to be retrieved</param>
+        /// <returns>A <see cref="T"/> object if the entity is found, otherwise null</returns>
+        Task<T> GetByIdAsync(Guid guid);

```

Comment:
¿every entity has a GUID id?

Reply:
Not sure about this, maybe I should let this method just in the UserService class

Reply:
Yes, or you can always parameterize the identity like IBaseService<TEntity, TIdentity>, and use it for the GetById(TIdentity)

Reply:
that sound much better to me, i will do that, thanks for the comments.

Reply:
Thank you for improving the seeder c:

Reply:
I would not do that, since BaseEntity is hardcoded with a Guid, I think it's safe to assume that a Guid is good enough.

We could make the improvement of having TIdentity generic as you suggested, but I think that might be outside of the scope of this request (also, it's a lot of work and I'm not sure we want to support those cases).

Reply:
oh, i see so the firts question's answer it's yes, so, you're right, there's no need to make it generic, Thank you @AlphaGit, i'll mark this as resolved then